File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/domains/HouseImage.java
package com.chaincuerealestate.chaincuerealestate.domains;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity(name = "house_image")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class HouseImage {
    @Id
    private String id;
    private String url;

    public static HouseImage create(String url) {
        return new HouseImage(
                UUID.randomUUID().toString(),
                url
        );
    }
}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/domains/HouseImage.kt
package com.example.chaincuerealestate.domains

import jakarta.persistence.Entity
import jakarta.persistence.Id
import java.util.UUID

@Entity(name = "house_image")
data class HouseImage(
    @Id
    var id: String,
    var url: String
) {
    companion object {
        fun create(url: String): HouseImage {
            return HouseImage(
                id = UUID.randomUUID().toString(),
                url = url
            )
        }
    }
}
